src/App.tsx

```diff
@@ -3,33 +3,56 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+  // movies: Movie[],
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+    // movies: [...moviesFromServer],
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState(() => ({
+      query: event.target.value,
+    }));
+  };
 
   render() {
+    const filteredMovies = moviesFromServer
+      .filter(movie => {
+        return (
+          movie.title.toLowerCase().includes(this.state.query.toLowerCase())
+          || movie.description.toLowerCase().includes(this.state.query.toLowerCase())
+        );
+      });
```

Comment:
You can short and optimize this:
```suggestion
      .filter({ title, description } => {
      	const lowerCaseQuery = this.state.query.toLowerCase();
      	
        return (
          title.toLowerCase().includes(lowerCaseQuery)
          || description.toLowerCase().includes(lowerCaseQuery)
        );
      });
```